indico/modules/receipts/client/js/printing/EventImageSelectField.jsx

```diff
@@ -0,0 +1,110 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Form, Header, HeaderContent, HeaderSubheader, Icon, IconGroup} from 'semantic-ui-react';
+
+import {Translate} from 'indico/react/i18n';
+
+import './EventImageSelectField.module.scss';
+
+const imageShape = {
+  identifier: PropTypes.string.isRequired,
+  filename: PropTypes.string.isRequired,
+  preview: PropTypes.string.isRequired,
+  supported: PropTypes.bool.isRequired,
+};
+
+const imageSource = {
+  images: Translate.string('from Images'),
+  attachments: Translate.string('from Materials'),
+  logo: Translate.string('from Logo'),
+};
+
+const EventImageOption = ({identifier, filename, preview, supported}) => (
+  <Header styleName="option-header">
+    <div
+      styleName="preview"
+      style={preview ? {backgroundImage: `url('data:image/jpeg;base64,${preview}')`} : undefined}
+    >
+      {!preview && (
+        <IconGroup size="big">
+          <Icon name="file image" />
+          <Icon corner="bottom right" name="dont" />
+        </IconGroup>
+      )}
+    </div>
+    <HeaderContent>
+      {filename} {!supported && <Translate as="em">(unsupported image type)</Translate>}
+      <HeaderSubheader content={imageSource[new URL(identifier).host]} />
+    </HeaderContent>
+  </Header>
+);
+
+EventImageOption.propTypes = imageShape;
+
+const EventImageSelectField = ({
```

Comment:
I'd use a normal `function EventImageSelectField({...}) { return ...; }` for this.

1) you can do `export default function ...`
2) I kind of hate expression-style arrow functions for react components if i need to add some console log when testing/debugging, since now it means i have to make many changes just to add a statement in there